The PR addresses an issue where re-initializing the library client in a notebook results in an `AttributeError` due to a missing `trace_store` attribute in the `TelemetryAdapter` object. The error occurs because certain required steps for the object state were bypassed as part of the global `_TRACE_PROVIDER` check. The proposed fix involves moving the initialization of the object state out of the `_TRACE_PROVIDER` initialization.